api/foundation/foundation_api.adoc

```diff
@@ -0,0 +1,141 @@
+
+There are two notable superclasses for a new plugin's activator class. They are `org.jboss.tools.foundation.core.plugin.BaseCorePlugin` and `org.jboss.tools.foundation.ui.plugin.BaseUIPlugin`. 
+A new plugin should extend one of these two classes in almost all cases unless prevented by other architectural concerns. 
+
+`BaseCorePlugin` and `BaseUIActivator` grant access to several very useful APIs. See the three following protected methods:
+
+```
+protected IPluginLog pluginLogInternal()        // To easily log messages
+protected StatusFactory statusFactoryInternal() // To create status objects
+protected void registerDebugOptionsListener(String pluginId, AbstractTrace trace, BundleContext bc) // To aid in tracing
+```
+
+Since these three methods are all protected, they are not callable from outside the class heirarchy. 
+A new plugin's activator should expose the plugin log and the status factory methods in whatever way
+they decide is best. Most often, with the goal of concise code for callers, a plugin activator 
+would probably expose these methods as static methods as follows:
+
+```
+	public static IPluginLog pluginLog() {
+		return getDefault().pluginLogInternal();
+	}
+	public static StatusFactory statusFactory() {
+		return getDefault().statusFactoryInternal();
+	}
+```
+
+== Tracing
+
+Tracing is a mechanism provided by eclipse to help log a subset of messages based on various keys. It is very useful when debugging, because you can 
+select which types of messages you want logged. In runtime-workbench, you can accomplish this via UI, and in a full installation, you can enable this
+by using a .options file in your eclipse installation's root folder. 
+
+For some information on eclipse tracing, see https://wiki.eclipse.org/FAQ_How_do_I_use_the_platform_debug_tracing_facility
+
+In `foundation`, however, we provide a base class `AbstractTrace` which you should extend, as well as a method to assist in wiring up the tracing. 
+
+=== Your Trace Class
+
+At a minimum, your Trace class should look as follows:
+
+```
+public class Trace extends AbstractTrace implements DebugOptionsListener {
+	private static Trace instance = null;
+	public Trace(StacksCoreActivator p) {
+		super(p);
+		instance = this;
+	}
+
+	public static void trace(final String level, String s) {
+		trace(level, s, null);
+	}
+
+	public static void trace(final String level, String s, Throwable t) {
+		traceInternal(instance, level, s, t);
+	}
+}
+```
+
+This allows very concise call methods later, where you can simply call
+
+    Trace.trace(Trace.STRING_EXTENSION_POINT, "Loading some ext. pt");
+
+The abstract super class includes several useful pre-defined trace levels:
+
+```
+	public final static String STRING_CONFIG = "/config"; //$NON-NLS-1$
+	public final static String STRING_INFO = "/info"; //$NON-NLS-1$
+	public final static String STRING_WARNING = "/warning"; //$NON-NLS-1$
+	public final static String STRING_SEVERE = "/severe"; //$NON-NLS-1$
```

Comment:
finest, finer, info, warning, severe   are simply 'levels' as opposed to 'categories'.  Developers should trace using 'finest' for the absolute greatest detail,  'finer' for things one step higher,   'info' for things more notable,  'warning' for things that may cause problems, and 'severe' for big or unexpected problems. 

Anything traced with 'severe' will automatically be added to the error log.